packages/dumb/src/components/settings/history/history-table-content.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import _ from 'lodash'
+import { HistorySpannedRow } from './history-table-spanned-row'
+import { HistoryTableRow } from './history-table-row'
+import React, { FunctionComponent } from 'react'
+import { getLongDayHourFormat } from '../../../utils/datetime/datetime.util'
+import moment from 'moment-timezone'
+import { TimePrefs, Unit } from 'medical-domain'
+import { ChangeType, HistorizedParameter, IncomingRow } from '../../../models/historized-parameter.model'
+
+interface HistoryTableContentProps {
+  onSwitchToDaily: Function
+  data: IncomingRow[]
+  timePrefs: TimePrefs
+  length: number
+}
+
+export const HistoryTableContent: FunctionComponent<HistoryTableContentProps> = (props): JSX.Element => {
+  const { onSwitchToDaily, data, timePrefs, length } = props
+
+  // TODO: Candidate to custom hooks ?
```

Comment:
Simply a `packages/dumb/src/components/settings/history/history-table-content.util.ts` that you could already test! :slightly_smiling_face:

Reply:
agree i think the same after reflexion